app/org/sagebionetworks/bridge/models/surveys/SurveyRule.java

```diff
@@ -15,75 +18,64 @@
         DE  // declined to answer
     }
     
-    private Operator operator;
-    private Object value;
-    private String skipToTarget;
+    private final Operator operator;
+    private final Object value;
+    // Rule should either have a skipToTarget, or have endSurvey set to true, but never both.
+    // This is enforced in rules validation.
+    private final String skipToTarget;
+    private final Boolean endSurvey;
     
-    public SurveyRule() {
-    }
-    public SurveyRule(Operator operator, Object value, String skipToTarget) {
+    @JsonCreator
+    private SurveyRule(@JsonProperty("operator") Operator operator, @JsonProperty("value") Object value,
+            @JsonProperty("skipTo") String skipToTarget, @JsonProperty("endSurvey") Boolean endSurvey) {
         this.operator = operator;
         this.value = value;
         this.skipToTarget = skipToTarget;
+        this.endSurvey = Boolean.TRUE.equals(endSurvey) ? Boolean.TRUE : null;
+    }
+    public SurveyRule(Operator operator, Object value, String skipToTarget) {
+        this(operator, value, skipToTarget, null);
+    }
+    public SurveyRule(Operator operator, Object value) {
+        this(operator, value, null, Boolean.TRUE);
     }
```

Comment:
Overloaded constructors can get really messy really fast. I strongly recommend using the builder pattern instead.